internal/exec/engine.go

```diff
@@ -272,7 +273,19 @@ func (e *Engine) fetchReferencedConfig(cfgRef types.ConfigReference) (types.Conf
 	if err != nil {
 		return types.Config{}, err
 	}
-	rawCfg, err := e.Fetcher.FetchToBuffer(*u, resource.FetchOptions{})
+	var headers http.Header
+	if cfgRef.HTTPHeaders != nil && len(cfgRef.HTTPHeaders) > 0 {
+		if u.Scheme != "http" && u.Scheme != "https" {
+			return types.Config{}, e.Logger.Crit("Cannot use HTTP headers with %v source scheme", u.Scheme)
```

Comment:
Hmm, I think we need to create a new error here. `e.Logger.Crit` just logs it, it doesn't return an error object wrapping the message passed.

Reply:
right, done